File: TicketMangementAPI/Models/Ticket.cs
namespace TicketsAPI.Models
{
    public class Ticket
    {
        public int Id { get; set; }
        public string ShortDescription { get; set; } = default!;
        public string Description { get; set; } = default!;
        public DateTime CreatedDate { get; set; }
        public string Severity { get; set; } = default!;
        public DateTime TargetDate { get; set; }
        public string Status { get; set; } = default!;
    }
}

File: ticket-client/src/app/services/ticket.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

export interface Ticket {
  id: number;
  shortDescription: string;
  description: string;
  createdDate: string; // ISO string from API
  severity: string;
  targetDate: string;
  status: string;
}

@Injectable({
  providedIn: 'root'
})
export class TicketService {
  private apiUrl = 'https://localhost:7077/api/Tickets'; // adjust to your API URL/port

  constructor(private http: HttpClient) {}

  getTickets(): Observable<Ticket[]> {
    return this.http.get<Ticket[]>(this.apiUrl);
  }
}
